react-validated-input/my-app/src/App.tsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { FaCheck } from 'react-icons/fa';
+
+import './App.css';
+import { IconType } from 'react-icons';
+
+function App() {
+  const [password, setPassword] = useState('');
+
+  function handlePassword(valid: string | IconType, notValid: string): string {
+    return password.length === 0
+      ? notValid
+      : password.length < 8
+      ? notValid
+      : /^(?=.*\d)(?=.*[A-Z])(?=.*[!@#$%^&*()]).+$/.test(password)
+      ? valid
+      : notValid;
+  }
+  return (
+    <>
+      <h3>Password</h3>
+      <input
+        type="password"
+        value={password}
+        onChange={(e) => setPassword(e.target.value)}></input>
+      <i className={handlePassword('green', 'red')}>
+        {handlePassword(<FaCheck />, ' x')}
+      </i>
+      <p className="error-message red">
+        {password.length === 0
```

Comment:
It seems like you should be able to combine this with the `handlePassword` you do above. That would eliminate some of the duplication.